_user_guides/armory-pipeline-templates.md

```diff
@@ -0,0 +1,189 @@
+Any number of variables can be overwritten in the same module by simply specifying them as arguments. e.g.: `{{ module "wait.stage.module" "waitTime" 100 "name" "simpleWait" }}`.
+
+> Note: We do not support complex data-type variable substitution in the alpha release
+
+Let us create a more realistic pipeline using templates. One that would look like this:
+
+![demopipeline](http://f.cl.ly/items/1z3z3Z2w3j2w35171U39/Screen%20Shot%202018-03-12%20at%2011.18.38%20AM.png)
+
+You would use the following JSON to create such. Note that any of the stages could have come from an imported module, but we show the full JSON here for readability:
+
+```
+{
+  "application": "demo",
+  "pipelines": [
+    {
+      "application": "demo",
+      "keepWaitingPipelines": false,
+      "limitConcurrent": true,
+      "name": "step1",
+      "stages": [
+        {
+          "continuePipeline": false,
+          "failPipeline": true,
+          "isNew": true,
+          "job": "armory/job/armory-hello-deploy/job/master",
+          "master": "Armory Jenkins",
+          "name": "Jenkins",
+          "parameters": {},
+          "refId": "105",
+          "requisiteStageRefIds": [],
+          "type": "jenkins"
+        },
+        {
+          "baseLabel": "release",
+          "baseOs": "ubuntu",
+          "cloudProviderType": "aws",
+          "extendedAttributes": {},
+          "isNew": true,
+          "name": "bake in eu-central-1",
+          "package": "myapp_1.27-h343",
+          "refId": "101",
+          "regions": [
+             "eu-central-1"
+          ],
+          "requisiteStageRefIds": [
+            "105"
+          ],
+          "storeType": "ebs",
+          "type": "bake",
+          "user": "LeSandeep",
+          "vmType": "hvm"
+        },
+        {
+          "failPipeline": true,
+          "isNew": true,
+          "name": "run tests",
+          "refId": "102",
+          "requisiteStageRefIds": [
+            "101"
+          ],
+          "type": "script",
+          "user": "LeSandeep",
+          "waitForCompletion": true
+        },
+        {
+          "isNew": true,
+          "name": "canary",
+          "refId": "103",
+          "requisiteStageRefIds": [
+            "101"
+          ]
+        },
+        {
+          "clusters": [],
+          "isNew": true,
+          "name": "deploy to stage",
+          "refId": "104",
+          "requisiteStageRefIds": [
+            "102",
+            "103"
+          ],
+          "type": "deploy"
+        }
+      ],
+      "triggers": []
+    }
+  ]
+}
+```
+
+
+## How it works in a nutshell
+
+GitHub (or BitBucket) webhooks are sent off when either the Templates or the Module definitions are modified. The Dinghy service looks for and fetches all dependent modules and parses the template and updates the pipelines in Spinnaker. The pipelines get automatically updated whenever a module that is used by a pipeline is updated in VCS. This is done by maintaining a dependency graph.
```

Comment:
I think this is pretty important, maybe move it after the intro section, before the primatives.

Reply:
>This is done by maintaining a dependency graph.
this sentence is a bit ambiguous, can you elaborate on it? 🙏